content/blogs/gui-setup-part2-partitioning.md

```diff
@@ -0,0 +1,88 @@
+The user then may feel more comfortable performing such operations by viewing the resulting changes: new partition layout, etc., knowing they haven't yet been actually applied, until his/her final choice has been made.
+This is why, for example, 3rd-party partitioning software allow to stash (queue) all these changes, allowing for a careful review before actually applying them.
+
+Because of these considerations, I chose to make the ReactOS GUI setup follow a similar philosophy.
+The partition page allows the user to choose the partition where to install ReactOS, as well as to modify the partition layout, keeping the changes in memory until the actual ReactOS installation is started.
+
+<<< IMAGE ROS GUI SETUP PARTITION >>>
+
+This new installation workflow necessitated to adapt the existing partitioning code, that is shared with the text-mode setup; to determine where and how to store additional per-partition arbitrary data state (for example, filesystem choice and formatting parameters).
+
+Most importantly, this required two main changes in the code.
+- Rewriting the formatter/chkdsk machine-state code, that was used in USETUP, to make it more generic and act as a queue whose stashed actions can be then committed at once.
+- splitting the bootloader installation choice, from its actual installation code.
+This was required, because the old behaviour was to present the user with the bootloader installation choice only at the very end of ReactOS installation, and not before starting it, and the old code was not reusable elsewhere.
+- better unify the helper functions used to create partitions.
+Existing code was differencing "primary", "logical" and "extended" partitions. This makes sense for MBR-based disks only, but does not anywhere else (GPT disks...).
+These partition creation helpers now automatically deduce their actual type -- the only exception being for "extended" partitions for MBR disks only.
+
+<<< IMAGE CREATE PARTITION DIALOG >>>
+
+
+## What's next -- partition-wise
+
+Partition handling code is now present in the GUI setup.
+It however requires now few improvements:
+- the possibility to select an empty disk space and still be able to install ReactOS on it, provided it is eligible for that (large enough, and that a partition entry can be created on it);
+- the possibility to reformat from the UI an _existing_ partition:
+this is not yet possible because the formatting options currently exist only in the Create-And-Format-Partition dialog;
+- show in the partition list the selected filesystem for a newly-created partition, instead of showing `"New (Unformatted)"`.
+
+----
+
+In the next [blog post](/blogs/gui-setup-part3-TBD-TBD), I will discuss about the other functionalities of the GUI setup, not pertaining to partitioning, as well as the problems I encountered once attempting to test for the first time the whole (still crude) ReactOS GUI installer.
```

Comment:
I would completely remove this link, and add it back in when you write the next one.

Reply:
I now have the other part 3 being written and will make another PR soon, but yes.